src/libraries/Microsoft.Extensions.Options/gen/CommonUtilities.cs

```diff
@@ -0,0 +1,83 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.Extensions.Options.Generators
+{
+    internal sealed class CommonUtilities
```

Comment:
Nit: Most "utilities" classes I'm aware of are typically collections of static helper methods. Perhaps it could be named differently to convey that it's an instance containing state?